### PR TITLE
Upgrade documentation for testers

### PR DESCRIPTION
**Pull request by GiudGiud**
- add custom cmp explanation to CSVDiff
- fix variable names in exodiff
- add custom cmp for exodiff in exodiff instead of for csvdiff in exodiff refs #19983

Ran into this while fixing the VTB. This could be a tad clearer

### CODE REVIEW

File: modules/doc/content/python/testers/Exodiff.md [hunk 1]
```diff
@@ -10,10 +10,10 @@ Complete documentation for the `Exodiff` utility may be found [here](https://san
 
 Test configuration options are added to the `tests` file.
 
-- `abs_tol`: sets an absolute tolerance, defaults to 1e-10. Both absolute and relative tolerances must
+- `abs_zero`: sets an absolute tolerance, defaults to 1e-10. Both absolute and relative tolerances must
   be met for a test to pass.
 
-- `rel_tol`: sets a relative tolerance. Defaults to 5.5e-6.
+- `rel_err`: sets a relative tolerance. Defaults to 5.5e-6.
```
Comment (cticenhour) [inline modules/doc/content/python/testers/Exodiff.md:16]:
```suggestion
- `rel_err`: sets a relative error tolerance. Defaults to 5.5e-6.
```

File: modules/doc/content/python/testers/Exodiff.md [hunk 2]
```diff
@@ -10,10 +10,10 @@ Complete documentation for the `Exodiff` utility may be found [here](https://san
 
 Test configuration options are added to the `tests` file.
 
-- `abs_tol`: sets an absolute tolerance, defaults to 1e-10. Both absolute and relative tolerances must
+- `abs_zero`: sets an absolute tolerance, defaults to 1e-10. Both absolute and relative tolerances must
   be met for a test to pass.
```
Comment (cticenhour) [inline modules/doc/content/python/testers/Exodiff.md:14]:
```suggestion
- `abs_zero`: sets an absolute zero cutoff used in comparisons, defaults to 1e-10.
```

File: modules/doc/content/python/testers/JSONDiff.md [hunk 3]
```diff
@@ -12,6 +12,11 @@ Test configuration options are added to the `tests` file.
 
 - `skip_keys`: A list of keys to skip in the `JSON` comparison
 
+- `abs_zero`: Sets an absolute tolerance, defaults to 1e-10. Both absolute and relative tolerances must
+  be met for a test to pass.
+
+- `rel_err`: Sets a relative tolerance, defaults to 5.5e-6.
+
```
Comment (cticenhour) [inline modules/doc/content/python/testers/JSONDiff.md:19]:
```suggestion
- `abs_zero`: Sets an absolute zero cutoff used in comparisons, defaults to 1e-10.

- `rel_err`: Sets a relative error comparison tolerance, defaults to 5.5e-6.

```

File: modules/doc/content/python/testers/Exodiff.md [hunk 4]
```diff
@@ -42,17 +42,13 @@ ${MOOSE_DIR}/framework/contrib/exodiff/exodiff -summary moose_simulation_out.e
 
 This generates a summary of the simulation which can be edited to make the
 custom comparison file, where the tolerances for each variable may be set individually.
-In the example shown below, the relative tolerance criterion for the `z_field` variable
-is modified from the global value of `5.5e-6` to `10`. This essentially disables
-comparison for that field.
+In the example below, the tolerances are adjusted for local comparisons of field variables,
+both nodal and elemental. The relative error criteria is set to `1e-6` and the absolute
+criteria is disabled by setting it to `0`. These modifications apply to variable `u` and `distance`,
```
Comment (cticenhour) [inline modules/doc/content/python/testers/Exodiff.md:47]: By "absolute criteria" do you mean the floor parameter? This effectively sets the absolute zero parameter for comparisons on a per group basis. Having it set to 0 is the exodiff default for variables and then 1e-15 for timesteps (per the [Exodiff doco](https://sandialabs.github.io/seacas-docs/exo_util.pdf#page=11)). 

If floor is what you were referring to, this statement should be tweaked.

**Review by cticenhour — CHANGES_REQUESTED** [review]
Thanks once again for the improvements here - a few comments.

File: modules/doc/content/python/testers/Exodiff.md (hunk 4 above)
Comment (GiudGiud) [inline modules/doc/content/python/testers/Exodiff.md:47]:
```suggestion
criteria is strongly tightened by setting it to `0`. These modifications apply to variable `u` and `distance`,
```

**Comment by GiudGiud** [comment]
should be ready for another look

**Review by cticenhour — APPROVED** [review]
Thanks for clarifying your intention with the phrasing in that follow-up commit - looks good now.

**Pull request merged**